File: Service/src/main/java/com/github/spb/tget/demo/repository/dbRepository/ClientDbRepository.java
package com.github.spb.tget.demo.repository.dbRepository;

import com.github.spb.tget.demo.data.Client;
import com.github.spb.tget.demo.repository.DbRepository;

public class ClientDbRepository extends DbRepository<Client> {

    private ClientDbRepository() {
        this.entityName = "Client";
    }

    public static ClientDbRepository create() {
        return new ClientDbRepository();
    }
}

File: Service/src/main/java/com/github/spb/tget/demo/repository/dbRepository/ClientDbRepository.kt
package com.github.spb.tget.demo.repository.dbRepository

import com.github.spb.tget.demo.data.ClientEntity

class ClientDbRepository private constructor() : DbRepository<ClientEntity>() {

    init {
        this.entityName = "ClientEntity"
    }

    companion object {

        fun create(): ClientDbRepository {
            return ClientDbRepository()
        }
    }
}